packages/render-fidelity-tools/src/artifact-creator.ts

```diff
@@ -48,18 +48,124 @@ export class ArtifactCreator {
         renderer => ({...renderer, file: `${renderer.name}-golden.png`}));
   }
 
-  async captureAndAnalyzeScreenshots(
-      scenarioWhitelist: Set<string>|null = null) {
+  async compareRenderers(scenario: ScenarioConfig) {
+    const analysisResults: AnalysisResults = [];
+    const {rootDirectory, outputDirectory, goldens} = this;
+    const {name: scenarioName, exclude, dimensions} = scenario;
+
+    console.log(
+        `Start to compare model-viewer's golden with other renderers' goldens:`);
+
+    const modelViewerIndex = 0;
+    const modelViewerGoldenPath = join(
+        rootDirectory, 'goldens', scenarioName, goldens[modelViewerIndex].file);
+    let modelViewerGolden;
+    try {
+      modelViewerGolden = await fs.readFile(modelViewerGoldenPath);
+    } catch (error) {
+      throw new Error(`❌ Failed to read model-viewer's ${
+          scenarioName} golden! Error message: ${error.message}`);
+    }
+
+    await fs.writeFile(
+        join(outputDirectory, scenarioName, goldens[modelViewerIndex].file),
+        modelViewerGolden);
+
+    const modelViewerGoldenImage = pngjs.PNG.sync.read(modelViewerGolden).data;
+
+    for (let golden of goldens) {
+      if (golden.name === 'model-viewer' ||
+          (exclude != null && exclude.includes(golden.name))) {
+        continue;
+      }
+
+      console.log(`\n🔍 Comparing <model-viewer> to ${golden.description}`);
+
+      const candidateGoldenPath =
+          join(rootDirectory, 'goldens', scenarioName, golden.file);
+      let candidateGolden;
+      try {
+        candidateGolden = await fs.readFile(candidateGoldenPath);
+      } catch (error) {
+        throw new Error(`❌ Failed to read ${golden.name}'s ${
+            scenarioName} golden! Error message: ${error.message}`);
+      }
+
+      await fs.writeFile(
+          join(outputDirectory, scenarioName, golden.file), candidateGolden);
+
+      const candidateGoldenImage = pngjs.PNG.sync.read(candidateGolden).data;
+      const analysisResult = await this.analyze(
+          modelViewerGoldenImage, candidateGoldenImage, dimensions, false);
+      analysisResults.push(analysisResult);
+    }
+    const scenarioRecord = {analysisResults, scenario};
+
+    console.log(`\n💾 Recording analysis`);
+    await fs.writeFile(
+        join(outputDirectory, scenarioName, 'analysis.json'),
+        JSON.stringify(scenarioRecord));
+  }
+
+  async autoTest(scenario: ScenarioConfig): Promise<ImageComparisonAnalysis> {
```

Comment:
autoTest is a pretty generic name. Let's try to be specific about what we're doing. Maybe this is the function that should be called `captureAndAnalyzeScreenshot`?